test/e2e/cpi_vm_test.go

```diff
@@ -199,10 +240,66 @@ var _ = Describe("Restarting VMs", func() {
 
 				Expect(newExternalIP).To(Equal(externalIP))
 				Expect(newInternalIP).To(Equal(internalIP))
-				Expect(getProviderIPFromNode(workerNode)).To(Equal(providerID))
+				Expect(getProviderIDFromNode(workerNode)).To(Equal(providerID))
 
 				return nil
 			}).Should(Succeed())
 		})
 	})
+
+	It("should result in new node when recreating VM", func() {
+
+		Eventually(func() bool {
+			workerNode, err = getWorkerNode()
+			if err != nil {
+				return false
+			}
+			return DoesNodeHasReadiness(workerNode, corev1.ConditionTrue)
+		}, 10*time.Minute).Should(BeTrue())
+
+		By("Read the externalIP, internalIP and providerID of VM")
+		externalIP, err := getExternalIPFromNode(workerNode)
+		Expect(err).ToNot(HaveOccurred())
+
+		internalIP, err := getInternalIPFromNode(workerNode)
+		Expect(err).ToNot(HaveOccurred())
+
+		providerID := getProviderIDFromNode(workerNode)
+
+		By("Delete machine object", func() {
+			err := deleteWorkerMachine(workerNode.Name)
+			Expect(err).To(BeNil(), "cannot delete machine object")
+		})
+
+		By("Eventually original node will be gone")
+		Eventually(func() bool {
+			_, err = getWorkerNode()
+			return err != nil && err.Error() == "worker node not found"
+		}, 5*time.Minute, 5*time.Second).Should(BeTrue())
+
+		By("Eventually new node will be created")
+		var newExternalIP, newInternalIP string
+		Eventually(func() error {
+			if workerNode, err = getWorkerNode(); err != nil {
+				return err
+			}
+			if newExternalIP, err = getExternalIPFromNode(workerNode); err != nil {
+				return err
+			}
+			if newInternalIP, err = getInternalIPFromNode(workerNode); err != nil {
+				return err
+			}
+			return nil
+		}, 10*time.Minute, 5*time.Second).Should(Succeed())
+
+		By("New node will be created with correct info, different from old one")
+		Expect(workerNode.Name).ToNot(Equal(originalWorkerNodeName), "name still the same")
+		Expect(newExternalIP).ToNot(Equal(externalIP), "newExternalIP still the same")
```

Comment:
IP could still be the same, that depends on DHCP server

Reply:
fixed